src/Infrastructure/Event/RequestTransformerListener.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Infrastructure\Event;
+
+use JsonException;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+
+class RequestTransformerListener
+{
+    /**
+     * @var array<string>
+     */
+    protected array $contentTypes;
+
+    /**
+     * @param array<string> $contentTypes
+     */
+    public function __construct(array $contentTypes)
+    {
+        $this->contentTypes = $contentTypes;
+    }
+
+    /**
+     * @param \Symfony\Component\HttpKernel\Event\RequestEvent $event
+     *
+     * @return void
+     */
+    public function onKernelRequest(RequestEvent $event): void
+    {
+        $request = $event->getRequest();
+
+        if ($this->supports($request) === false) {
+            return;
+        }
+
+        try {
+            $data = json_decode((string)$request->getContent(), true, 512, \JSON_THROW_ON_ERROR);
+
+            if (is_array($data)) {
```

Comment:
It will be always an array, 'cause the second parameter is `true`. this check can be removed.

Reply:
`var_dump(json_decode('null', true, 512, \JSON_THROW_ON_ERROR));` ===> null

Reply:
fair enough